src/url/queryRebuild.js

```diff
@@ -0,0 +1,33 @@
+
+import d3 from 'd3';
+
+/**
+ * Takes query and returns an object of query values that adhere to the
+ * following conditions:
+ * - the parameter is an urlKey attribute of one of the values in urlQueryConfig
+ * - the urlQueryConfig for the parameter is marked with persist = true.
+ *
+ * Persistance defaults to true - so the absense of a persist flag is the same
+ * as persist = true.
+ * @param {Object} urlQuery Query object from something location.query
+ * @param {Object} urlQueryConfig URL configuration object with object values
+ * including `urlKey`.
+ * @return {Object} rebuilt query Object only including persistent values.
+ */
+export default function queryRebuild(urlQuery, urlQueryConfig) {
+  const rebuiltQuery = {};
+
+  const configValues = d3.values(urlQueryConfig);
+
+  configValues.forEach((config) => {
+    // persist defaults to true.
+    if ((config.persist === undefined) || (config.persist)) {
```

Comment:
I think the simpler way of saying this is `config.persist !== false` (which means null, undefined, and true will work.. just have to explicitly set to `false` to avoid)